paper/paper.md

```diff
@@ -0,0 +1,84 @@
+
+When used with JupyterHub [@JupyterHub], nbgrader provides additional workflow functionality that covers the entire grading process. After creating the assignment, instructors can release it to students, who can then fetch a copy of the assignment directly through the notebook server interface. Students can submit their completed version through the same interface, making it available for instructors to collect with one command. After that, instructors can use the auto-grading functionality as normal and may access the formgrader as a JupyterHub service.
+
+Since its conception in September 2014, nbgrader has been "battle-tested" in a number of classes all over the world, including at UC Berkeley, Cal Poly San Luis Obispo, University of Pretoria, University of Edinburgh, Northeastern University, Central Connecticut State University, KTH Royal Institute of Technology Stockholm, CU Boulder, University of Amsterdam, George Washington University, Texas A&M University, Bryn Mawr College, and University of Maryland; and, as of May 2018, over 10,000 nbgrader-based notebooks exist on GitHub. In addition to its core functionality, nbgrader has expanded to support a number of features, including the ability to handle multiple classes on the same JupyterHub instance; the option to either include or hide autograder tests; customizable late penalties; and support for importing assignment files downloaded from a LMS (Learning Management System). As we continue to develop nbgrader, we always keep it’s original aim in mind: to provide a flexible, straightforward system for creating and grading assignments in the Jupyter notebook.
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields outside of computer science. As these disciplines require more computational tools, undergraduate curricula also begin to include topics in programming and computer science. However, because students are focused on their own discipline—and programming is likely a secondary interest—teaching students through traditional computer science offerings is not always effective [@Cortina2007; @Forte2005; @Guzdial2005]. While there are visual programming languages such as Raptor [@Carlisle2004] or Scratch [@Resnick2009] that are intended to be easy and enjoyable for non-computer science majors to learn, they lack the specialized tools that are required for effective work in domain sciences, such as numerical or data visualization libraries. A hybrid approach is to teach students computational concepts in an interactive environment where it is possible to quickly write, test, and tweak small pieces of code. Many such environments exist, including Mathematica [@mathematica], Maple [@maple], Matlab [@matlab], Sage [@sage] and IPython [@PerezGranger2007].
```

Comment:
- [ ] Citation needed for opening sentence.
- [ ] "more computational tools" is ambiguous: are the tools becoming more computational, or are there more of them.
- [ ] Rather than call it "their own discipline", which reads mildly as a statement of ownership, you might call it something like the discipline that is the object of their study.
- [ ] `computer science offerings` => `computer-science offerings`
- [ ] Raptor *and* Scratch
- [ ] `non-computer science majors` => `non computer-science majors`
- [ ] This reads as though the libraries are examples of the domain sciences: "...the specialized tools that are required for effective work in domain sciences, such as numerical or data visualization libraries". How about "...specialized tools such as numerical or data visualization libraries, which are required for effective work in domain sciences." Domain sciences might be a good substitute for what I suggested above regarding "their own".
- [ ] `pieces` => `units` or `samples` or `bits`? Pieces feels slightly unidiomatic, as though I might have to pick them up off the floor.
- [ ] MATLAB is written in all caps.

Reply:
I've addressed these, except for the suggestions to rename "computer science" to "computer-science", because "computer science" as a term does not typically have a hyphen (in the same way that "earth science" does not).

> The use of computational methods has become increasingly widespread in fields outside of computer science.

I'm not sure what the best citation for this would be, any suggestions? It is just generally true that there are a large number of computational fields now outside of computer science (e.g. computational biology, computational neuroscience, data science, etc.) and the demand on computer science departments to teach intro programming classes is huge. I am sure there are papers discussing this, but I am not sure if there's a canonical citation or not.

Reply:
How about changing "non computer-science majors" to "those unfamiliar with programming"? That's a way to write around the issue ("science majors who are not computers"), while expanding the scope a bit to better capture Scratch et al.'s intended market.

Reply:
Regarding citations, maybe this? http://rsta.royalsocietypublishing.org/content/366/1881/3717.short